ts/Series/Treemap/TreemapSeries.ts

```diff
@@ -682,22 +858,41 @@ class TreemapSeries extends ScatterSeries {
 
             // If options for level exists, include them as well
             if (level?.dataLabels) {
-                options = merge(options, level.dataLabels);
+                merge(true, options, splat(level.dataLabels)[0]);
                 series.hasDataLabels = (): boolean => true;
             }
 
+            // Headers are always top-aligned. Leaf nodes no not support
+            // headers.
+            if (point.node.isLeaf) {
+                options.inside = true;
+            } else if (options.headers) {
+                options.verticalAlign = 'top';
+            }
+
             // Set dataLabel width to the width of the point shape minus the
             // padding
             if (point.shapeArgs) {
-                const css = {
-                    width: (
-                        (point.shapeArgs.width || 0) -
-                        2 * (options.padding || padding || 0)
-                    ) + 'px',
-                    lineClamp: Math.floor((point.shapeArgs.height || 0) / 16)
-                };
-                extend((options.style as any), css);
-                point.dataLabel?.css(css);
+                const { height = 0, width = 0 } = point.shapeArgs;
+                if (width > 32 && height > 16 && point.shouldDraw()) {
```

Comment:
Should those static numbers be configurable, or should they depend on font size?